src/main/java/com/gtnewhorizons/modularui/api/NumberFormatMUI.java

```diff
@@ -0,0 +1,218 @@
+     * calls to this baseFormat, but when calling code changes any of the parameters (e.g., number of decimal places),
+     * we store the changes in this instance of NumberFormatMUI. If the player changes their locale in the config, we
+     * create a new baseFormat from the new locale, and use the stored parameters to also update the new baseFormat.
+     * This means that the calling code can keep only one instance of NumberFormatMUI, configure its parameters, and
+     * this one instance will automatically be updated if the player's locale changes.
+     */
+    protected DecimalFormat baseFormat;
+    protected Locale currentLocale;
+
+    public NumberFormatMUI() {
+        refreshBaseFormat();
+    }
+
+    /**
+     * Updates the base format based on the locale from ModularUI's config, and then modifies all properties that have
+     * been previously changed through this instance.
+     */
+    protected void refreshBaseFormat() {
+        currentLocale = Config.locale;
+        NumberFormat nf = NumberFormat.getNumberInstance(currentLocale);
+
+        if (nf instanceof DecimalFormat) {
+            baseFormat = (DecimalFormat) nf;
+        } else {
+            // Current locale does not provide a DecimalFormat.
+            baseFormat = (DecimalFormat) NumberFormat.getNumberInstance(Locale.US);
+        }
+
+        // Workaround: The grouping separator in French locales is the non-breakable space '\u00A0',
+        // which does not display well in the Minecraft font. Replace it with a plain space.
+        DecimalFormatSymbols dfs = baseFormat.getDecimalFormatSymbols();
+        if (dfs.getGroupingSeparator() == '\u00A0' || dfs.getGroupingSeparator() == '\u202F') {
+            dfs.setGroupingSeparator(' ');
+            baseFormat.setDecimalFormatSymbols(dfs);
+        }
+
+        // Update any fields that have been previously changed via NumberFormatMUI.
+        if (minimumIntegerDigitsChanged) baseFormat.setMinimumIntegerDigits(getMinimumIntegerDigits());
+        if (maximumIntegerDigitsChanged) baseFormat.setMaximumIntegerDigits(getMaximumIntegerDigits());
+        if (minimumFractionDigitsChanged) baseFormat.setMinimumFractionDigits(getMinimumFractionDigits());
+        if (maximumFractionDigitsChanged) baseFormat.setMaximumFractionDigits(getMaximumFractionDigits());
+
+        if (groupingUsedChanged) baseFormat.setGroupingUsed(isGroupingUsed());
+        if (parseIntegerOnlyChanged) baseFormat.setParseIntegerOnly(isParseIntegerOnly());
+        if (roundingModeChanged) baseFormat.setRoundingMode(getRoundingMode());
+    }
+
+    /* Formatting / Parsing */
+
+    protected static final FieldPosition unusedFieldPosition = new FieldPosition(0);
+
+    public StringBuffer format(double number, StringBuffer toAppendTo) {
+        if (currentLocale != Config.locale) refreshBaseFormat();
+        return baseFormat.format(number, toAppendTo, unusedFieldPosition);
```

Comment:
`return format(number, toAppendTo, unusedFieldPosition)`?

Reply:
It would just be one more unnecessary method call that eventually does exactly the same thing.

Reply:
Well I just prefer this way, but I won't block it